### PR TITLE
ability to customize User-Agent header in the HTTP requests sent by cURL

### PR DESCRIPTION
**Pull request by samuelecat**
Ability to customize User-Agent header in the HTTP requests sent by cURL fetching a content on remote server.
This allows the owners of the remote server to customize the response.

### CODE REVIEW

File: src/RemoteContentFetcher.php [hunk 1]
```diff
@@ -31,7 +31,11 @@ public function getFileContentsByCurl($url)
 
 		$ch = curl_init($url);
 
-		curl_setopt($ch, CURLOPT_USERAGENT, 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:13.0) Gecko/20100101 Firefox/13.0.1'); // mPDF 5.7.4
+		if ($this->mpdf->curlUserAgent) {
+			curl_setopt($ch, CURLOPT_USERAGENT, $this->mpdf->curlUserAgent);
+		} else {
+			curl_setopt($ch, CURLOPT_USERAGENT, 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:13.0) Gecko/20100101 Firefox/13.0.1'); // mPDF 5.7.4
+		}
```
Comment (MarkVaughn) [inline src/RemoteContentFetcher.php:38]:
```suggestion
		curl_setopt($ch, CURLOPT_USERAGENT, $this->mpdf->curlUserAgent);	
```

File: src/Config/ConfigVariables.php [hunk 2]
```diff
@@ -514,6 +514,7 @@ public function __construct()
 			'curlTimeout' => 5,
 			'curlProxy' => null,
 			'curlProxyAuth' => null,
+			'curlUserAgent' => null,
```
Comment (MarkVaughn) [inline src/Config/ConfigVariables.php:517]:
```suggestion
			'curlUserAgent' => 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:13.0) Gecko/20100101 Firefox/13.0.1',
```

Comment (finwe) [reply src/Config/ConfigVariables.php:517]: Why this UA in particular?

**Comment by finwe** [comment]
I'd rather extract the remote content fetching to an interface that could be set up and replaced altogether. Preferably in 9.0.

**Comment by samuelecat** [comment]
> suggest moving the default user agent into the config

done :)

**Comment by samuelecat** [comment]
It would be helpful for me to have this small feature in your next release in the way I don't have to reference my forked repo, is there anything I can do about it? thank you

**Comment by finwe** [comment]
Please add a Changelog entry and squash the commits to one.

**Comment by samuelecat** [comment]
> Please add a Changelog entry and squash the commits to one.

done, thank you

**Pull request merged**